Documentation for `tab.group`, `tab.ungroup`, and additions of `groupId` (#39145)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/tabs/ungroup/index.md

---
title: tabs.ungroup()
slug: Mozilla/Add-ons/WebExtensions/API/tabs/ungroup
page-type: webextension-api-function
browser-compat: webextensions.api.tabs.ungroup
---

{{AddonSidebar}}

> Removes one or more tabs from their respective groups. If any groups become empty, they are deleted.

Reviewer: ```suggestion
Removes one or more tabs from their respective tab groups. If any groups become empty, the empty tab groups are deleted.

If a tab is between other tabs of the same tab group before ungrouping, the tab may be moved to ensure that the remaining tabs of the group stay adjacent to each other.

Note: the `tabs.ungroup()` method is not the only way to ungroup tabs; a tab may also be ungrouped when it is moved by calling {{WebExtAPIRef("tabs.move")}} with an `index` away from the tab group.
```

Besides the explanation I expanded "they" so that the reader cannot mistake a group for a tab.

## Syntax

```js-nolint
let ungrouping = browser.tabs.ungroup(
  tabIds              // array
)
```

### Parameters

- `tabIds`

>   - : `array`. The tab ID or list of tab IDs to remove from groups.

Reviewer: ```suggestion
  - : `integer` or `array` of `integer`. The tab ID or list of tab IDs to remove from groups.
```

### Return value

> A [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that is resolved with no arguments.

Reviewer: ```suggestion
A [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that is resolved with no arguments.

If any of the `tabIds` are invalid, the promise is rejected without modifying the tabs.
```

## Examples

> Remove two tabs from their tab groups:
> 
> ```js
> function onRemoved() {
>   console.log(`Tabs removed from groups`);
> }
> 
> function onError(error) {
>   console.log(`Error: ${error}`);
> }
> 
> let removing = browser.tabs.ungroup( tabIds: [2, 13] );
> removing.then(onRemoved, onError);

Reviewer: Extensions can never input a number and expect a meaningful result. tabIds are always retrieved from other APIs, so here is an example:

```suggestion
Remove a tab from their tab group, if any:

```js
let tabs = await browser.tabs.query({
  active: true,
  lastFocusedWindow: true,
});

await browser.tabs.ungroup([tabs[0].id]);
console.log("Current tab is ungrouped");
```

---
title: tabs.ungroup()
slug: Mozilla/Add-ons/WebExtensions/API/tabs/ungroup
page-type: webextension-api-function
browser-compat: webextensions.api.tabs.ungroup
---

{{AddonSidebar}}

Removes one or more tabs from their respective tab groups. If any groups become empty, they are deleted.

> If a tab is between other tabs in a tab group before ungrouping, the remaining tabs are moved so they stay adjacent.

Reviewer: Your rephrased version of https://github.com/mdn/content/pull/39145#discussion_r2046476407 changed the meaning subtly. The tabs to ungroup move, not the other tabs.

Here is a new suggestion:

```suggestion
If a tab is between other tabs in a tab group before ungrouping, the tab moves outside the group to ensure that the remaining tabs stay adjacent.
```

Reviewer: Here is an alternative suggestion, to match my suggested text in the `group()` method:

```suggestion
All tabs in a tab group must be adjacent to each other. An ungrouped tab is moved before or after the tab group if needed to maintain this requirement.
```

Author: ```suggestion
All tabs in a tab group must be adjacent. If necessary, an ungrouped tab is moved before or after the tab group to maintain this requirement.
```

> [!NOTE]

> > The `tabs.ungroup()` method is not the only way to ungroup tabs. A tab is ungrouped when it's moved by calling {{WebExtAPIRef("tabs.move")}} with an `index` that is outside the tab group.

Reviewer: When I read the text, the second sentence may give the impression that the `tabs.move()` method has to be called to achieve the objective of ungrouping. To avoid that mistaken interpretation, how about adding `"also"`?

```suggestion
> The `tabs.ungroup()` method is not the only way to ungroup tabs. A tab is also ungrouped when it's moved by calling {{WebExtAPIRef("tabs.move")}} with an `index` that is outside a tab group.
```
